### PR TITLE
Null Geodesics in Kerr (& Schwarzschild)

### PR DESCRIPTION
**Pull request by JeS24**
ToDo:

- [x]  Structure the modules in `geodesic` and `rays`.
- [x] Implement Null Geodesic calculation for arbitrary **starting** positions & momentum.
- [x] Move to Hamiltonian Dynamical Approach, using Julia. Write a wrapper around it.
- [x] Move Julia code to `einsteinpy_geodesics`. Add tests, docs, CI.
- [x] Make `geodesic` compatible with `einsteinpy_geodesics` (Julia back-end). Also, implement a Python back-end for fallback.
- [x] Add tests.
- [x] Add warning for python back-end usage.
- [x] Edit notebooks, examples. ([Link](https://gfycat.com/famousapprehensiveheterodontosaurus) to a constant radius orbit, animated using new code )
- [x] Remove artefacts.
- [x] Edit CI config to (mock)test (parts of) Julia back-end. All of the main tests, that involve `julia` are now in the `einsteinpy_geodesics` repo.
- [x] Drop Python 3.6.
- [x] Write tests for `Christoffels` & `f_vec` methods in the metric classes
- [x] Update `CHANGELOG`.

@shreyasbapat

### CODE REVIEW

File: src/einsteinpy/geodesic/__init__.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+from .geodesic import Geodesic
```
[the remark this answers is not in the document]
Comment (shreyasbapat) [reply src/einsteinpy/geodesic/__init__.py:1]: Read this as the oldest comment. I am not this happy as of now xD

**Comment by shreyasbapat** [comment]
Complete the final bit and we will merge. Then let's drop every other plan and work on making this better

File: src/einsteinpy/geodesic/utils.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+        (a ** 2 + (-2 + q[0]) * q[0])
+        * (a ** 2 + 2 * q[0] ** 2 + a ** 2 * np.cos(2 * q[1]))
+    )
+
+
+def _sphToCart(r, th, ph):
+    """
+    Utility function to convert Spherical Polar
+    Coordinates to Cartesian Coordinates
+
```
Comment (rishi-s8) [inline src/einsteinpy/geodesic/utils.py:78]: Could you add `Parameters` and `Returns` to the docstring?

File: src/einsteinpy/geodesic/utils.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+    return yn
+
+
+def _verlet_step(ld, y, params):
+    """
+    Synchornized Symplectic VerletLeapfrog Integrator
+    Advances integration by one step
+
+    Source: `Wikipedia: <https://en.wikipedia.org/wiki/Leapfrog_integration#Algorithm>`_
+
```
Comment (rishi-s8) [inline src/einsteinpy/geodesic/utils.py:151]: `Parameters` and `Returns`

**Review by rishi-s8 — COMMENTED** [review]
Looks good from my side. :) Just a change in one of the docstrings.

File: src/einsteinpy/geodesic/utils.py (hunk 2 above)
Comment (JeS24) [reply src/einsteinpy/geodesic/utils.py:78]: Sure. Done.

**Comment by shreyasbapat** [comment]
Of Course it will fail of Python 3.6

**Comment by shreyasbapat** [comment]
`einsteinpy-geodesics` is a Python 3.7+ package

**Comment by shreyasbapat** [comment]
Let's drop Python 3.6 from EinsteinPy

**Comment by JeS24** [comment]
> `einsteinpy-geodesics` is a Python 3.7+ package
> 

Oh, snap. Yeah, it is.

**Comment by shreyasbapat** [comment]
Okay no

**Comment by shreyasbapat** [comment]
I am making a commit

**Comment by shreyasbapat** [comment]
NO DROP IT

**Comment by JeS24** [comment]
Will fix all the PEP issues in the morning, except W503, of course.

**Comment by JeS24** [comment]
All issues fixed.

File: src/einsteinpy/symbolic/predefined/vacuum_solutions.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def ReissnerNordstorm(
     sch=symbols("r_s"),
     Q=symbols("Q"),
 ):
-    """"
+    """ "
```
Comment (shreyasbapat) [inline src/einsteinpy/symbolic/predefined/vacuum_solutions.py:125]: What's this?

**Comment by shreyasbapat** [comment]
I have some code quality reservations, but I will let it go because of time constraint

File: src/einsteinpy/symbolic/predefined/vacuum_solutions.py (hunk 4 above)
Comment (JeS24) [reply src/einsteinpy/symbolic/predefined/vacuum_solutions.py:125]: @shreyasbapat The extra quotation mark? Seems like, it was mistyped. Removing it.

Comment (JeS24) [reply src/einsteinpy/symbolic/predefined/vacuum_solutions.py:125]: There was an `isort` update, which broke the quality checks (https://app.circleci.com/pipelines/github/einsteinpy/einsteinpy/1885/workflows/130a38b8-2db8-4b63-a0a1-fc0cf7fdeadd/jobs/8730). I guess, while making the changes then, I mistyped this. And, `isort` update is also why, there are many extraneous edits in this PR.

Comment (JeS24) [reply src/einsteinpy/symbolic/predefined/vacuum_solutions.py:125]: Oh, well. I have some edits queued in the `warp` branch, that adds the Alcubierre Metric. I will push this change there (#533).

**Pull request merged**